barretenberg/cpp/src/barretenberg/commitment_schemes/zeromorph/zeromorph.hpp

```diff
@@ -694,26 +704,102 @@ template <typename PCS> class ZeroMorphVerifier_ {
 
         // Compute commitment C_{\zeta,Z}
         Commitment C_zeta_Z;
+        FF evaluation;
         if constexpr (Curve::is_stdlib_type) {
             // Express operation as a batch_mul in order to use Goblinization if available
             auto builder = z_challenge.get_context();
             std::vector<FF> scalars = { FF(builder, 1), z_challenge };
             std::vector<Commitment> points = { C_zeta_x, C_Z_x };
             C_zeta_Z = Commitment::batch_mul(points, scalars);
+            evaluation = FF::from_witness(builder, 0);
         } else {
             C_zeta_Z = C_zeta_x + C_Z_x * z_challenge;
+            evaluation = FF(0);
         }
 
-        // Define the evaluation (always zero by construction in this case) for the PCS opening
-        FF evaluation{ 0 };
-        if constexpr (Curve::is_stdlib_type) { // add builder if in circuit context
-            auto builder = z_challenge.get_context();
-            evaluation = FF(builder, 0);
+        return { .opening_pair = { .challenge = x_challenge, .evaluation = evaluation }, .commitment = C_zeta_Z };
+    }
+
+    /**
+     * @brief Verify a set of multilinear evaluation claims for unshifted polynomials f_i and to-be-shifted
+     * polynomials g_i
+     *
+     * @param commitments Commitments to polynomials f_i and g_i (unshifted and to-be-shifted)
+     * @param claimed_evaluations Claimed evaluations v_i = f_i(u) and w_i = h_i(u) = g_i_shifted(u)
+     * @param multivariate_challenge Challenge point u
+     * @param transcript
+     * @return VerifierAccumulator Inputs to the final PCS verification check that will be accumulated
+     */
+    static VerifierAccumulator verify(RefSpan<Commitment> unshifted_commitments,
```

Comment:
I really wanted to use the PCS inside Zeromorph to be consistent with the paper (as ZM is supposed to be instantiated with a univariate commitment schemes) and  overloading seemed the best option. We really dont want to make kzg verification take the verifier commitment key before the pairing check because that means adding it to the recursive verifier when there is no need for it (recursive verifiers dont do the pairing check).